### ISSUE TITLE
trie-store integrity check is running for 9h already - Please reconsider integrity checks after a crash

### ISSUE THREAD

**Issue by GuybrushX** [casper-network/casper-node#2268]
I'm in the situation that I migrated a testnet node to another server which went well. As you probably know the testnet DB has ~500GB already because of some "load tests". After the migration I started the node on the new server and everything seemed to be fine -> it was syncing. Unfortunately another "load test" was started and the new node seemed to be stuck for about 2h (no log entries anymore, iostat also didn't show any activity anymore) -> I decided to restart the casper-node-launcher via systemctl restart casper-node-launcher. That seemed to kill the node without a clean shutdown and the LMDB was in a dirty state.

That was now 9h ago and it is still "running trie-store integrity check, this may take a while". iotop shows almost constantly 2.15 G/s DISK READ.

I found the following comment:
https://github.com/casper-network/casper-node/blob/release-1.4.1/node/src/reactor/initializer.rs#L234

Are there any plans/possibilities to improve that kind of checks in the new future? Maybe split the DB into smaller chunks? According to @sacherjj 1.4.1 will flush the DB more often (if I understood him correctly) but not sure if that will help with such checks.

Besides the unacceptable long time it would also be very nice to see some kind of progress since I have absolutely no idea what the node is doing and if it checked 95% already or only 3% or something in between.

If I imagine that something like this can happen easily to a mainnet validator I'm really afraid since that means losing lots of delegators, rewards and generally big trouble.

**Comment by GuybrushX** [comment]
Quick update:
The integrity check completed successfully after over 10h:
{"timestamp":"Oct 25 10:20:29.378","level":"WARN","fields":{"message":"queue sealed, dropping item"},"target":"casper_node::utils::round_robin"}
{"timestamp":"Oct 25 10:21:14.434","level":"WARN","fields":{"message":"queue sealed, dropping item"},"target":"casper_node::utils::round_robin"}
{"timestamp":"Oct 25 10:21:26.722","level":"WARN","fields":{"message":"queue sealed, dropping item"},"target":"casper_node::utils::round_robin"}

-> lots of those "queue sealed" messages which suddenly stopped and after 2h I issued systemctl restart casper-node-service.

{"timestamp":"Oct 25 12:21:42.562","level":"INFO","fields":{"message":"read stored state","path":"/etc/casper/casper-node-launcher-state.toml"},"target":"casper_node_launcher::launcher"}
{"timestamp":"Oct 25 12:21:42.568","level":"INFO","fields":{"message":"node starting up","version":"1.3.4-0d28d49"},"target":"casper_node::cli"}
{"timestamp":"Oct 25 12:21:42.568","level":"WARN","fields":{"message":"previous node instance seems to have crashed, integrity checks may be run"},"target":"casper_node::cli"}
{"timestamp":"Oct 25 12:21:42.568","level":"WARN","fields":{"message":"large event size, consider reducing it or boxing","event_size":"160","type_name":"\"casper_node::reactor::initializer::Event\""},"target":"casper_node::reactor"}
{"timestamp":"Oct 25 12:21:42.570","level":"INFO","fields":{"message":"reindexing block store"},"target":"casper_node::components::storage"}
{"timestamp":"Oct 25 12:22:00.652","level":"INFO","fields":{"message":"block store reindexing complete"},"target":"casper_node::components::storage"}
{"timestamp":"Oct 25 12:22:00.652","level":"INFO","fields":{"message":"initializing block body database"},"target":"casper_node::components::storage"}
{"timestamp":"Oct 25 12:22:01.301","level":"INFO","fields":{"message":"block body database initialized"},"target":"casper_node::components::storage"}
{"timestamp":"Oct 25 12:22:01.301","level":"INFO","fields":{"message":"initializing block metadata database"},"target":"casper_node::components::storage"}
{"timestamp":"Oct 25 12:38:52.277","level":"INFO","fields":{"message":"block metadata database initialized"},"target":"casper_node::components::storage"}
{"timestamp":"Oct 25 12:38:52.277","level":"INFO","fields":{"message":"initializing deploy metadata database"},"target":"casper_node::components::storage"}
{"timestamp":"Oct 25 12:39:15.793","level":"INFO","fields":{"message":"deploy metadata database initialized"},"target":"casper_node::components::storage"}
{"timestamp":"Oct 25 12:39:15.799","level":"INFO","fields":{"message":"running trie-store integrity check, this may take a while"},"target":"casper_node::reactor::initializer"}
{"timestamp":"Oct 25 23:01:07.611","level":"INFO","fields":{"message":"reactor main loop is ready"},"target":"casper_node::reactor"}
{"timestamp":"Oct 25 23:01:07.622","level":"INFO","fields":{"message":"valid run after an unplanned shutdown before upgrade due","current_chainspec_activation_point":"era 2005","highest_block":"executed block 628e..6919, parent hash 3514..a0b3, post-state hash b6e3..23a3, body hash 8273..1ee5, random bit false, timestamp 2021-10-25T06:21:00.800Z, era_id 2384, height 244594, protocol version: 1.3.4, era_end: era_report: era end: slash , reward PubKey::Ed25519(01a4..feac): 9325668412, PubKey::Ed25519(028e..1bc7): 8444805209 "},"target":"casper_node::components::chainspec_loader"}

-> I've omitted lots of more "PubKey" items.

Luckily the load tests seem to be stopped, too so the node was finally able to catch up and is fully synced and active again.

I think it all happened because the systemctl restart did forcefully exit the process after timing out.

I saved the full ~723MB (63MB compressed) log file in case somebody wants to investigate that issue further.
